language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -242,32 +292,57 @@ substitution, anything).
 -- 2. calls 'returnCek' on values ('TyAbs', 'LamAbs', 'Constant')
 -- 3. returns 'EvaluationFailure' ('Error')
 -- 4. looks up a variable in the environment and calls 'returnCek' ('Var')
+
+getArgsCount :: Builtin ann -> CekM uni Int
+getArgsCount (BuiltinName _ name) =
+    pure $ builtinNameAritiesIncludingTypes ! name
```

Comment:
Huh, I was going to comment that I didn't see how the array would give us efficient access. I didn't realise you could index into them with arbitrary enumerable things, or am I missing something?

Reply:
Arbitrary `Ix` things and we now derive `Ix` for `BuiltinName`.

Reply:
What's going to happen when everything's dynamic and we don't know the names of the built-ins at compile time?

Reply:
We store it alongside the dynamic name meanings, which we already have to fetch out to evaluate it. Now, making that more efficient is another problem...